### PR TITLE
API-8300: Insuranceplan post stub

### PR DESCRIPTION
**Pull request by jabando-va**

### CODE REVIEW

**Comment by jabando-va** [comment]
adding tests

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/insuranceplan/R4SiteInsurancePlanController.java [hunk 1]
```diff
@@ -62,6 +71,20 @@ private void insuranceFileOrDie(String id, RecordCoordinates recordCoordinates)
...
+      value = "/site/{site}/r4/InsurancePlan",
+      consumes = {"application/json", "application/fhir+json"})
+  public void insurancePlanCreate(
+      @Redact HttpServletResponse response,
+      @PathVariable(value = "site") String site,
+      @Redact @RequestBody InsurancePlan body) {
+    body.meta(Meta.builder().source("s1").build());
+    var newResourceUrl = linkProperties.r4().readUrl(site, "InsurancePlan", "{new-resource-id}");
+    response.setStatus(201);
+    response.addHeader("Location", newResourceUrl);
```
Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/insuranceplan/R4SiteInsurancePlanController.java:85]: `HttpHeaders.LOCATION`

File: vista-fhir-query/src/test/java/gov/va/api/health/vistafhirquery/service/controller/insuranceplan/R4SiteInsurancePlanControllerTest.java [hunk 2]
```diff
@@ -40,6 +43,20 @@ private RpcInvocationResultV1 _invocationResult(Object value) {
...
   }
 
+  @Test
+  void create() {
+    var response = new MockHttpServletResponse();
+    var samples = InsurancePlanSamples.VistaLhsLighthouseRpcGateway.create();
+    var results = samples.getsManifestResults();
+    var captor = requestCaptor(LhsLighthouseRpcGatewayCoverageWrite.Request.class);
+    var answer =
+        answerFor(captor).value(results).invocationResult(_invocationResult(results)).build();
```
Comment (jhulbert-va) [inline vista-fhir-query/src/test/java/gov/va/api/health/vistafhirquery/service/controller/insuranceplan/R4SiteInsurancePlanControllerTest.java:53]: You don't need any of this yet.

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/insuranceplan/R4SiteInsurancePlanController.java [hunk 3]
```diff
@@ -62,6 +72,20 @@ private void insuranceFileOrDie(String id, RecordCoordinates recordCoordinates)
...
 
+  @Override
+  @PostMapping(
+      value = "/site/{site}/r4/InsurancePlan",
+      consumes = {"application/json", "application/fhir+json"})
+  public void insurancePlanCreate(
+      @Redact HttpServletResponse response,
+      @PathVariable(value = "site") String site,
+      @Redact @RequestBody InsurancePlan body) {
+    body.meta(Meta.builder().source("s1").build());
```
Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/insuranceplan/R4SiteInsurancePlanController.java:83]: ?

File: vista-fhir-query/src/test/java/gov/va/api/health/vistafhirquery/service/controller/insuranceplan/R4SiteInsurancePlanControllerTest.java [hunk 4]
```diff
@@ -40,6 +42,15 @@ private RpcInvocationResultV1 _invocationResult(Object value) {
         .build();
   }
 
+  @Test
+  void create() {
+    var response = new MockHttpServletResponse();
+    _controller()
+        .insurancePlanCreate(
+            response, "123", InsurancePlanSamples.R4.create().insurancePlan("123", "ip1"));
+    assertThat(response.getStatus()).isEqualTo(201);
```
Comment (jhulbert-va) [inline vista-fhir-query/src/test/java/gov/va/api/health/vistafhirquery/service/controller/insuranceplan/R4SiteInsurancePlanControllerTest.java:51]: Assertion for the location header as well.

**Review by jhulbert-va — APPROVED** [review]

**Review by bschofield-va — APPROVED** [review]

**Pull request merged**